### ISSUE TITLE
Change build_scream.cmake to include a mach file matching mach-compiler combo

### ISSUE THREAD

**Issue by bartgol** [E3SM-Project/scream#1665]
By including `components/scream/cmake/machine-files/${MACH}.cmake`, we complicate our lives, because we need the mach file to switch between compiler choices. Instead, we should include `${MACH}_${COMPILER}.cmake`. For instance, on Summit, this would make switching between P9 and V100 much easier, with `summit_gnugpu.cmake` containing all v100/cuda options, and `summit_gnu.cmake` contianing all p9/openmp options.

**Comment by jgfouca** [comment]
The latest improvements to the machine file system should make this very easy. Quartz is already doing this. The compiler-specific file should include the compiler-neutral file:

`include(${CMAKE_CURRENT_LIST_DIR}/${machine_name}.cmake)`

Which will set SCREAM_MACHINE to ${machine_name} and set compiler agnostic things. The neutral file should always have this at the top:

```
include(${CMAKE_CURRENT_LIST_DIR}/common.cmake)
common_setup()
```

I will make it so the E3SM build system will consider a mach_compiler file above a plain file.

**Comment by sarats** [comment]
We are at a point where SCREAM and E3SM machine configurations have very few (if any) conflicts left. Perhaps it would be better to converge to the same system now rather than maintaining two variations of each machine.

My 2 dollars.

**Comment by jgfouca** [comment]
@sarats , the issue is that we want to continue to support scream standalone which means no CIME and no config_machines.xml.

**Comment by jgfouca** [comment]
@sarats , there's a few things the need to happen. I plan to work with @bartgol to make more improvements to E3SM's build system, some of which could facilitate making life easier for components that want standalone capabilities.

The first thing that must be done is to transition the cmake macros under cime_config to use cmake names for things (example CMAKE_F90_FLAGS instead FFLAGS). Once that's done, we should be pretty close on the build side. That would leave mpirun and batch stuff, which involves fairly complex logic in CIME to compute.

**Comment by sarats** [comment]
Yes, looks like everyone is sort of managing their own machine files/flags. Don't folks use other components (land, ice,...) in standalone mode without CIME? I see lot of I-cases in PACE, so perhaps they just use CIME infrastructure to build.

Btw, probably not a concern but standalone runs won't have any provenance captured.

**Comment by bartgol** [comment]
@rljacob tbc, standalone scream does more than just running full atm. We have plenty of unit tests, that run small chunks of code. Stuff like
- while F90 and CXX are both alive, ensure the two impl's are BFB (at least without opt flags on)
- check functionalities of infrastructure classes
- check properties of certain parametrizations. TBH, I'm not sure how many we have of these, but should be stuff like "check X output is always non-negative, or monotone, or is div-free, or has property Y".

Then we also have full atm tests. For those, we are completely ignoring surf coupling (and read some "valid" surf field from IC files). These tests are more for checking that all the plumbing is in place, rather than doing BFB against baselines (although we do BFB checks across number of ranks). Adding baselines tests for standalone atm runs is on the todo list, but it's low priority, since we also run CIME tests, which already cover that.

In short, standalone tests are helpful while doing development and/or debug, since they provide smaller reproducers (or, if passing, ensure that the bug is more related to the interplay between atm processes and/or other components).

**Comment by rljacob** [comment]
Standalone usually means you're not using the driver or any other components as well as doing your own build.   Land only I-cases have a way of bypassing the coupler to read forcing data faster but otherwise build within CIME and use cpl7/mct.

**Comment by bartgol** [comment]
Right, that's what our standalone tests do: no component coupler, no CIME. Forcing data for full atm data (in standalone mode) is read in along with initial conditions.
